### PR TITLE
fix:[NEXT-347] improve multiple record handling

### PR DESCRIPTION
**Pull request by kevin-paladin**
Improve multi-record handling by caching with keys (data source and tenant id)
Improved messaging around existing files and merge results

## Summary by CodeRabbit

- **New Features**
	- Enhanced cache management in the AssetGroups class for improved data consistency.
	- Introduced logging enhancements across multiple classes for better traceability and clarity during asset processing.

- **Bug Fixes**
	- Improved error handling and logging in the Assets class to summarize asset processing outcomes effectively.

- **Chores**
	- Removed unnecessary dependencies in method signatures across various classes to streamline functionality.
	- Deleted the AssetGroupsRepository interface, simplifying asset group management operations.

### CODE REVIEW

File: assets-management/services/svc-asset-delta-engine/test-data/gcp-all.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         {
             "messageId": "059f36b4-87a3-44ab-83d2-661975830a7d",
             "receiptHandle": "AQEBwJnKyrHigUMZj6rYigCgxlaS3SLy0a...",
-            "body": "--tenant-id=saasdev --data-source=gcp --s3-path=mapper-results/saasdev/gcp/20240930-174929",
+            "body": "--tenant_id=98c28482-9bae-46bd-bd4e-58fa132e72c0 --data_source=gcp --s3_path=mapper-results/98c28482-9bae-46bd-bd4e-58fa132e72c0/gcp/20241108-040600 --skip_asset_count=true",
```
Comment (plyubich) [inline assets-management/services/svc-asset-delta-engine/test-data/gcp-all.json:6]: I suggest that in the future, we throw test files that contain actual values to the ivan repo. Although, this is not sensitive data, we should clean it up in code repos to dummy values when checking in.

**Review by plyubich — CHANGES_REQUESTED** [review]
@kevin-paladin - please see my comments

File: assets-management/services/svc-asset-delta-engine/src/main/java/com/paladincloud/common/config/ConfigService.java [hunk 2]
```diff
@@ -12,12 +12,16 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
 import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
 
 public class ConfigService {
 
+    private static final Logger LOGGER = LogManager.getLogger(ConfigService.class);
+    private static String propertiesTenantId = null;
```
[the remark this answers is not in the document]
Comment (kevin-paladin) [reply assets-management/services/svc-asset-delta-engine/src/main/java/com/paladincloud/common/config/ConfigService.java:24]: If we were multi-threaded, it'd be a necessity. But this is a single-threaded lambda, there's nothing allowing parallel access to this class.

File: assets-management/services/svc-asset-delta-engine/src/main/java/com/paladincloud/common/config/AssetTypes.java [hunk 3]
```diff
@@ -23,6 +23,7 @@ public class AssetTypes {
 
     private static final Logger LOGGER = LogManager.getLogger(AssetTypes.class);
     private static Map<String, Map<String, String>> typeInfo;
+    private static String typeInfoDataSource = null;
```
[the remark this answers is not in the document]
Comment (kevin-paladin) [reply assets-management/services/svc-asset-delta-engine/src/main/java/com/paladincloud/common/config/AssetTypes.java:26]: This is the same - we aren't in a multi-threaded environment. If we were, I'd change to the ConcurrentHashMap

File: assets-management/services/svc-asset-delta-engine/test-data/gcp-all.json (hunk 1 above)
Comment (kevin-paladin) [reply assets-management/services/svc-asset-delta-engine/test-data/gcp-all.json:6]: Well, that's a brilliant idea. I wish I'd thought of it.

**Review by plyubich — DISMISSED** [review]

**Pull request merged**